edweiss-app/components/lectures/remotecontrol/modal.tsx

```diff
@@ -519,3 +520,116 @@ export const QuestionBroadcastModal: ReactComponent<{
         </ModalContainer>
     );
 };
+
+
+
+// ------------------------------------------------------------
+// ---------------      Quiz Broadcast Modal     --------------
+// ------------------------------------------------------------
+
+export const QuizBroadcastModal: ReactComponent<{
+    modalRef: React.RefObject<BottomSheetModalMethods>;
+    id: string;
+    quizModel?: LectureQuizzes.LectureQuiz;
+    courseId: string,
+    lectureId: string,
+    broadcasted: string,
+    setBroadcasted: React.Dispatch<React.SetStateAction<string>>;
+    onClose: () => void;
+}> = ({ modalRef, id, courseId, lectureId, quizModel, broadcasted, setBroadcasted, onClose }) => {
+
+
+    // Hooks
+    const [broadLoading, setBroadLoading] = useState(false);
+    const [showLoading, setShowLoading] = useState(false);
+    const [showResultToStudent, setShowResultToStudent] = useState(false);
+
+    useEffect(() => {
+        if (quizModel?.showResultToStudents !== undefined) {
+            setShowResultToStudent(quizModel.showResultToStudents);
+        }
+    }, [quizModel?.showResultToStudents]);
+
+
+    async function toggleResult() {
+        setShowLoading(true);
+        const res = await callFunction(Quizzes.Functions.toggleLectureQuizResult, { lectureId: lectureId, lectureEventId: id, courseId: courseId, });
+
+        if (res.status === 1) {
+            console.log(`toggled showResultToStudent boolean`);
+        } else {
+            console.log(`Error while toggling boolean shoresultToStudent`);
+        }
+        setShowResultToStudent(!showResultToStudent)
+        setShowLoading(false);
+    }
+
+
+
+    async function handleQuizBroadcast() {
+        if (broadcasted === id) {
+
+            setBroadLoading(true);
+
+            // End of quiz activity
+            try {
+
+                await callFunction(LectureDisplay.Functions.markEventAsDone, {
+                    courseId: courseId,
+                    lectureId: lectureId,
+                    id: id,
+                });
+
+                await callFunction(LectureDisplay.Functions.clearQuestionEvent, {
+                    courseId: courseId,
+                    lectureId: lectureId,
+                    id: id,
+                });
+            } catch (error) { console.error("Error updating the quiz event:", error); }
+
+            setBroadcasted("");
+            setBroadLoading(false);
+            onClose();
```

Comment:
What does this do?

Reply:
Alright, so the first one is a hook passed by the modal from the `quizToSlide` screen. It's stores the `id` of the currently `broadcasted` activity. In case it's empty this means that no activity is currently broadcasted.  As you can see  we are in the branch to terminate the activity after the callfunctions have sucessfully called, so it's reset the current broadcasted to 'none'.
The second, is a hook that I use for the loading effect of the button, as the previous one, once I am sure that it `onPress` stuff are done, I can toggle off it state.
The last one is the call of the `onClose` method pass as an argument. It's contain the code that needs to be executed on exit off the modal.

Tell me if you need more informations.

Reply:
Oh okay, I did not notice the onClose method reference so I was a bit confused thinking it was a react native thing